element-hq/element-x-android
# Restore intentional mentions in the markdown/plain text editor

## Content

- Add `TextPillificationHelper` to add pills to existing MD text by using regexes searching for raw ids and links pointing to permalinks with ids.
- Use the helper when loading a draft, setting the text from the edit mode, or loading the MD version of the rich text content when toggling format.
- Split `MentionSpanProvider` into `MentionSpanProvider` and `MentionSpanTheme`, which contains the info about colors, typography and spacing to use in `MentionSpan`.
- Modify `MentionSpan` so those values can be update after the span creation.

## Motivation and context

Fixes https://github.com/element-hq/element-x-android/issues/2966

## Screenshots / GIFs

No UI changes, I just renamed a preview.

## Tests



- In a room, without formatting enabled, write `@room and @someotheruser`.
- Enable formatting, check the pills still work.
- Disable formatting, the pills should still be there.
- Go back to the room list, then back to the room. The draft should contain the pills too.
- Send the message, check the pills are rendered fine.
- Edit the message, check the pills are added correctly to the editor and once edited the message renders fine.

## Tested devices

- OS version(s): 14

## Checklist
